tonic-build/src/server.rs

```diff
@@ -257,92 +263,61 @@ fn generate_trait_methods<T: Service>(
             quote!(&self)
         };
 
-        let method = match (
```

Comment:
I'm not sure if the benefits of adding this feature are worth the increased complexity that comes with adding more features. However, the refactoring of this part that was done to add this feature seems to improve the readability and maintainability of the code, so if it be separated into a separate pull request, I think we can proceed with the discussion separately.

Reply:
Sure, the refactoring is in a separate commit so I'll put it in its own PR. Thanks!